source/data-formats/dates-and-times.txt

```diff
@@ -0,0 +1,448 @@
+      decoded = decode(val, codec_options=codec_ms)
+      print(decoded)
+
+   .. output::
+      :language: javascript
+
+      {"date": DatetimeMS(0)}
+
+DatetimeConversion.DATETIME_AUTO
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This value instructs {+driver-short+} to return ``datetime`` objects if the value is
+within ``datetime`` range, and return a ``DatetimeMS`` object otherwise. The following
+code example encodes and decodes one datetime in the ``datetime`` range and one
+outside the range:
+
+.. io-code-block::
+
+   .. input::
+      :language: python
+   
+      from datetime import datetime
+      from bson import encode, decode
+      from bson.datetime_ms import DatetimeMS
+      from bson.codec_options import CodecOptions, DatetimeConversion
+   
+      in_range = encode({"date": datetime(1970, 1, 1)})
+      out_of_range = encode({"date": DatetimeMS(-(2**62))})
+      codec_auto = CodecOptions(datetime_conversion=DatetimeConversion.DATETIME_AUTO)
+   
+      in_decoded = decode(in_range, codec_options=codec_auto)
+      out_decoded = decode(out_of_range, codec_options=codec_auto)
+      print(f"in-range date: {in_decoded}") 
+      print(f"out-of-range date: {out_decoded}") 
+
+   .. output::
+      :language: shell
+
+      in-range date: {"date": datetime.datetime(1970, 1, 1, 0, 0)}
+      out-of-range date: {'x': DatetimeMS(-4611686018427387904)}
+
+DatetimeConversion.DATETIME_CLAMP
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This value "clamps" the resulting ``datetime`` objects, forcing them to be within
+the ``datetime`` range (trimmed to 999,000 microseconds).
+
+The following code example encodes and decodes one datetime earlier than the ``datetime``
+range and one datetime later than the ``datetime`` range. The resulting values are
+at the beginning and end of the allowed range.
+
+.. io-code-block::
+
+   .. input::
+      :language: python
+
+      from datetime import datetime
+      from bson import encode, decode
+      from bson.datetime_ms import DatetimeMS
+      from bson.codec_options import CodecOptions, DatetimeConversion
+   
+      before = encode({"date": DatetimeMS(-(2**62))})
+      after = encode({"date": DatetimeMS(2**62)})
+      codec_clamp = CodecOptions(datetime_conversion=DatetimeConversion.DATETIME_CLAMP)
+   
+      before_decoded = decode(before, codec_options=codec_clamp)
+      after_decoded = decode(after, codec_options=codec_clamp)
+      print(f"datetime before the range: {before_decoded}")
+      print(f"datetime after the range: {after_decoded}")
+
+   .. output::
+      :language: shell
+
+      datetime before the range: {"date": datetime.datetime(1, 1, 1, 0, 0)}
+      datetime after the range: {"date": datetime.datetime(9999, 12, 31, 23, 59, 59, 999000)}
+
+.. tip::
+
+   ``DatetimeMS`` objects support rich comparison methods against other instances of
+   ``DatetimeMS``. You can also convert them to ``datetime`` objects by using
+   the ``DatetimeMS.to_datetime()`` method.
```

Comment:
This tip feels out of place here since it's not related specifically to the `DATETIME_CLAMP` enum. I think it might be better to remove it altogether, or potentially try to work it in to the paragraph where `DatetimeMS` objects are introduced

Reply:
good call